src/main/java/com/deque/axe/android/colorcontrast/ColorContrastRunner.java

```diff
@@ -140,6 +150,96 @@ ColorContrastResult onRowEnd() {
     return result;
   }
 
+  ColorContrastResult onRowEnd(AxeColor actualTextColor) {
+
+    ColorContrastResult result = new ColorContrastResult();
+
+    CountMap<ColorPair> pairsWithSimilarTextColor = new CountMap<>();
+
+    for (Map.Entry<ColorPair, Integer> entry1 : countExactPairs.entrySet()) {
+      for (Map.Entry<ColorPair, Integer> entry2 : countExactPairs.entrySet()) {
+        if (entry1.getKey().backgroundColor.equals(entry2.getKey().backgroundColor)) {
+          if (entry1.getKey().textColor.isVisiblySameColor(entry2.getKey().textColor)) {
+            pairsWithSimilarTextColor.increment(entry1.getKey(), entry2.getValue());
+          }
+        }
+      }
+    }
```

Comment:
I can but I chose not to because I first want to collect all the color combinations. In future if I want to make more adjustments then I need all the color combinations that we are able to gather from the that row of pixels. This logic also gives me an idea how correct the logic is when it is trying to gather color combinations.

Reply:
This is a really good question. Thanks for asking this 👍

Reply:
Maybe this is silly, but could this logic be pulled out into a function that then you can call when debugging?  It seems like it could be adding a lot of unnecessary complexity.  I'm not sure about Android, but for iOS we can automatically default to running a function in debug mode with compiler flags.  It just seems silly to run a lot of unnecessary code in production when it's only used for debug purposes?

Reply:
How about I create a separate issue for this? We already have to get rid of some old logic. I can clean this up in that issue, if you agree.

Reply:
Sure that makes sense to me!  That does seem outside of the scope of this PR :)

Reply:
Doh - just saw these comments lolol.

I echo these sentiments. Is there an issue already filed to address this complexity? 
I'm wondering on effort for this to just address it now and it can be QA'd together. If it's a bigger effort, I'm ok with breaking it out into a new issue.

Reply:
Yes, I have an issue in backlog to address all of the above. I agree that this PR is adding to the complexity that we already have but for a bigger refactor I need more time because the debugging process for this algorithm is not straight forward. In order to debug this I have to create a separate utility method to draw a red color line through the rows of pixels we are using to calculate the color contrast and create an image for the same and then look into each and every color combination that was used to get the result.